app/database/database.py

```diff
@@ -71,12 +73,23 @@ def add(self, record: Base):
         self.session.add(record)
         self.session.commit()
 
-    def find_all_plant_types(self, limit: int) -> List[PlantType]:
+    def get_all_plant_types(self, limit: int) -> List[PlantType]:
         query = select(PlantType).limit(limit)
         result = self.session.scalars(query)
         return result
 
-    def find_plant_type_by_botanical_name(
+    def get_logs_between(self,
+                         user_id: int,
+                         cleft: date,
+                         cright: date) -> List[Log]:
+        query = select(Log).\
+                where(Log.created_at.between(cleft, cright)).\
+                where(Log.plant.has(Plant.id_user == user_id)).\
```

Comment:
Me rompe esto:

plants  |   File "/app/app/controller/log_controller.py", line 34, in get_logs_by_user
plants  |     return req.app.database.get_logs_between(user_id, left, right)
plants  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
plants  |   File "/app/app/database/database.py", line 87, in get_logs_between
plants  |     where(Log.plant.has(Plant.id_user == user_id)).\
plants  |           ^^^^^^^^^
plants  | AttributeError: type object 'Log' has no attribute 'plant'

Hasta borre todo los contendores y volumenes de docker para que se cree de nuevo la tabla de la DB, y nada..
No se si en database.py quedo mal definida la relación

Reply:
Esto me ocurre cuando busco por `year=2024` y `user_id=1`

Reply:
Al remover la relacion del Log con la Planta fallaba esa parte. Encontre la forma de arreglarla sin que al traerme una planta me traiga todos los logs.